ci/infra/testrunner/platforms/terraform.py

```diff
@@ -115,10 +115,18 @@ def _generate_tfvars_file(self):
             with open(tfvars_final, "w") as f:
                 json.dump(tfvars, f)
 
+    #take up to 50 characters from stackname to give room to the fixed part
+    # in the node name: caasp-[master|worker]-xx (total length must be <= 63)
+    def stack_name(self):
+         stack_name = self.conf.terraform.stack_name
+         while len(stack_name) > 45:
```

Comment:
question. could we not remove this while loop in favor of just getting the sub string at length 45. And then if it is important to break it at a `-` just run the `rfind('-')` once after?

My thought is this. what if there is 59 characters before the first `-`? this would loop forever right? hopfully that cant happen, right?

Reply:
Good point. The intention of this loop is to remove whole components of the name. Truncating may lead to weird names.